dart/test/monster_test_my_game.example_generated.dart

```diff
@@ -387,6 +466,65 @@ class Vec3 {
   String toString() {
     return 'Vec3{x: $x, y: $y, z: $z, test1: $test1, test2: $test2, test3: $test3}';
   }
+
+  Vec3T unPack() => Vec3T(
+      x: x,
+      y: y,
+      z: z,
+      test1: test1,
+      test2: test2,
+      test3: test3.unPack());
+
+  static int pack(fb.Builder fbBuilder, Vec3T object) {
+    if (object == null) return 0;
+    return object.pack(fbBuilder);
+  }
+}
+
+class Vec3T {
+  double x;
+  double y;
+  double z;
+  double test1;
+  Color test2;
+  TestT test3;
+
+  Vec3T({
+      this.x,
+      this.y,
+      this.z,
+      this.test1,
+      this.test2,
+      this.test3});
+
+  int pack(fb.Builder fbBuilder) => _pack(fbBuilder, HashMap<Object, int>());
+
+  /// Write the object to the [Builder] only once and reuse the same offset on
+  /// subsequent references of the same object.
+  ///
+  /// Note that this method assumes you call it using the same [Builder]
+  /// instance every time. The returned offset is only good for the [Builder]
+  /// used in the first call to this method.
+  int packOnce(fb.Builder fbBuilder, Map<Object, int> existingOffsets) {
+    assert(existingOffsets != null);
+    return existingOffsets[this] ??= _pack(fbBuilder, existingOffsets);
+  }
+
+  int _pack(fb.Builder fbBuilder, Map<Object, int> existingOffsets) {
+    assert(existingOffsets != null);
+    assert(fbBuilder != null);
+
+    fbBuilder.pad(2);
+    test3.pack(fbBuilder);
+    fbBuilder.pad(1);
```

Comment:
the fact that its manually padding to me means that it is serializing manually rather than use the methods from the base API to serialize which presumably already pad for you? better to use those?

Reply:
Yet again, code from the existing ObjectBuilder.finish() - should - do you want to touch that, i.e. should I have a look why it doesn't use some existing method there?

Reply:
well, we shouldn't copy existing code from the base API, we should be calling into it.

Reply:
It is [existing generated code](https://github.com/google/flatbuffers/pull/6682/files#diff-7071eda2fbeaaac74ad8e19422a2a62af1b1dc931b49079665f8a93d154e7e3fR563-R574), not FlatBuffers API, and unfortunately can't be reused directly (except on the generator side, where I do it) because of a couple of differences, like how inner tables/structs are provided. I can do that but I'd probably have to change some of the generated code unrelated to ObjectAPIs - is that how you'd prefer to do this instead?